internal/gitstream/assign.go

```diff
@@ -0,0 +1,118 @@
+package gitstream
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-logr/logr"
+	"github.com/google/go-github/v47/github"
+	"github.com/qbarrand/gitstream/internal"
+	"github.com/qbarrand/gitstream/internal/config"
+	gh "github.com/qbarrand/gitstream/internal/github"
+	"github.com/qbarrand/gitstream/internal/gitutils"
+	"github.com/qbarrand/gitstream/internal/markup"
+	"gopkg.in/yaml.v3"
+)
+
+type Assign struct {
+	GC             *github.Client
+	DryRun         bool
+	Finder         markup.Finder
+	GitHelper      gitutils.Helper
+	Logger         logr.Logger
+	IssueHelper    gh.IssueHelper
+	UserHelper     gh.UserHelper
+	Repo           *git.Repository
+	RepoName       *gh.RepoName
+	UpstreamConfig config.Upstream
+}
+
+func (a *Assign) Run(ctx context.Context) error {
+	const remoteName = internal.UpstreamRemoteName
+
+	if _, err := a.GitHelper.RecreateRemote(ctx, remoteName, a.UpstreamConfig.URL); err != nil {
+		return fmt.Errorf("could not recreate remote: %v", err)
+	}
+
+	if err := a.GitHelper.FetchRemoteContext(ctx, remoteName, a.UpstreamConfig.Ref); err != nil {
+		return fmt.Errorf("could not fetch remote %s: %v", remoteName, err)
+	}
+
+	if err := a.assingIssuesAndPRs(ctx); err != nil {
+		return fmt.Errorf("could not add assignees to PRs/Issues: %v", err)
+	}
+
+	return nil
+}
+
+func (a *Assign) getOwnersContent(ctx context.Context) (*gh.Owners, error) {
+
+	const filePath = "OWNERS"
+
+	content, _, _, err := a.GC.Repositories.GetContents(ctx, a.RepoName.Owner, a.RepoName.Repo, filePath, nil)
```

Comment:
Can't we read the file from the local repo? When GitStream runs in CI, it has a local clone of the downstream repo.

Reply:
I guess we can but how do we mock it in UT?

Reply:
Looking how we fetch the u/s repo to see if it is an easy change.

Reply:
I will need to dig a bit more into the code to understand how it can be done.

Are you OK with merging this PR as is and I will create an issue to try to make it better? This way we can also check if it works.

Reply:
https://github.com/qbarrand/gitstream/issues/28